### ISSUE TITLE
Consumer core gives different NoC dst for matmul

### ISSUE THREAD

**Issue by yugaoTT** [tenstorrent-metal/tt-metal#5189]
To reproduce:
1. go to `yugao/wh_block_matmul_hang_consumer `
2. run `pytest tests/tt_eager/python_api_testing/unit_testing/test_bert_ops.py`
3. In `tt_metal/impl/dispatch/kernels/command_queue_consumer.hpp`, I added DPRINT info for `command_ptr[3], noc_x_start, noc_y_start, noc_x_end, noc_y_end`. The command_ptr[3] should contain the NoC dst.

The branch is based on the `yugao/wh_block_matmul_hang`, so it will han after running the test for around 100 iterations.

Machine IP: 172.27.44.23, harvested X2

**Comment by davorchap** [comment]
@yugaoTT is this a blocker?

**Comment by yugaoTT** [comment]
@davorchap no I was debugging the matmul_hang, and saw the consumer is mcasting go signal to a NoC dst of (1,3),(9,9) for matmul, while for interleaved_to_sharded the consumer is mcasting to (1,2),(9,9), which is a grid of 8x7.

**Comment by DrJessop** [comment]
I looked into this, and for matmul, given it is broken into two kernel groups, it is correct that Yu would see two GO signal mcasts. One potential bug we did catch though is that there is nothing stalling the dispatcher on sending the GO signals in the case where we have multiple kernel groups. That means it is possible for semaphores to not arrive to one group before another group gets its GO signal. Apparently this is not done for this matmul kernel, but it seems like a general problem. Ideally, we'd want to stall sending the GO signal before binaries/semaphores have arrived in cases where we have two kernel groups. @pgkeller  @davorchap FYI.
